numpy/ma/core.pyi

```diff
@@ -260,16 +262,18 @@ _ArrayT = TypeVar("_ArrayT", bound=ndarray[Any, Any])
 _ScalarT = TypeVar("_ScalarT", bound=generic)
 _ScalarT_co = TypeVar("_ScalarT_co", bound=generic, covariant=True)
 _NumberT = TypeVar("_NumberT", bound=number)
+_RealNumberT = TypeVar("_RealNumberT", bound=floating | integer)
 # A subset of `MaskedArray` that can be parametrized w.r.t. `np.generic`
 _MaskedArray: TypeAlias = MaskedArray[_AnyShape, dtype[_ScalarT]]
 
 _MaskedArrayUInt_co: TypeAlias = _MaskedArray[unsignedinteger | np.bool]
 _MaskedArrayInt_co: TypeAlias = _MaskedArray[integer | np.bool]
-_MaskedArrayComplex_co: TypeAlias = _MaskedArray[inexact | integer | np.bool]
-_MaskedArrayTD64_co: TypeAlias = _MaskedArray[timedelta64 | integer | np.bool]
 _MaskedArrayFloat64_co: TypeAlias = _MaskedArray[floating[_64Bit] | float32 | float16 | integer | np.bool]
-_MaskedArrayComplex128_co: TypeAlias = _MaskedArray[number[_64Bit] | number[_32Bit] | float16 | integer | np.bool]
 _MaskedArrayFloat_co: TypeAlias = _MaskedArray[floating | integer | np.bool]
+_MaskedArrayComplex_co: TypeAlias = _MaskedArray[inexact | integer | np.bool]
```

Comment:
they're pretty much the same:

```suggestion
_MaskedArrayComplex_co: TypeAlias = _MaskedArrayNumber_co
```

Reply:
tried this, but it fails
```
assert_type(MAR_number - AR_number, MaskedArray[np.number])
```

Reply:
Huh, `_MaskedArrayComplex_co` isn't even used..?